### Pemeriksaan Kualitas Data

In [14]:
df.info()                 # tipe data setiap kolom
df.isna().sum()           # jumlah data kosong per kolom
df.duplicated().sum()     # jumlah baris duplikat

<class 'pandas.DataFrame'>
RangeIndex: 103205 entries, 0 to 103204
Data columns (total 28 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   NAME                103205 non-null  str    
 1   HUMIDITY            98084 non-null   float64
 2   LIGHT               97133 non-null   float64
 3   NO_MAX              103205 non-null  int64  
 4   NO_MIN              103205 non-null  int64  
 5   NO2_MAX             102628 non-null  float64
 6   NO2_MIN             102628 non-null  float64
 7   OZONE_MAX           102597 non-null  float64
 8   OZONE_MIN           102597 non-null  float64
 9   PM10_MAX            99972 non-null   float64
 10  PM10_MIN            99972 non-null   float64
 11  PM2_MAX             99972 non-null   float64
 12  PM2_MIN             99972 non-null   float64
 13  SO2_MAX             102368 non-null  float64
 14  SO2_MIN             102368 non-null  float64
 15  CO_MAX              102597 non-null  float64


np.int64(0)

In [15]:
df.isna().sum()       # jumlah kosong per kolom
(df.isna().sum() / len(df) * 100).round(2) # dalam bentuk persen

NAME                   0.00
HUMIDITY               4.96
LIGHT                  5.88
NO_MAX                 0.00
NO_MIN                 0.00
NO2_MAX                0.56
NO2_MIN                0.56
OZONE_MAX              0.59
OZONE_MIN              0.59
PM10_MAX               3.13
PM10_MIN               3.13
PM2_MAX                3.13
PM2_MIN                3.13
SO2_MAX                0.81
SO2_MIN                0.81
CO_MAX                 0.59
CO_MIN                 0.59
CO2_MAX                1.52
CO2_MIN                1.52
SOUND                  4.96
TEMPRATURE_MAX         4.90
TEMPRATURE_MIN         4.90
UV_MAX                12.13
UV_MIN                12.13
AIR_PRESSURE           4.96
LASTUPDATEDATETIME     0.00
Lattitude              0.00
Longitude              0.00
dtype: float64

### Temuan Data

| Temuan | Keputusan | Alasan |
|---|---|---|
| Missing Values pada kolom polutan/cuaca (<13%per kolom | Dibiarkan | Presentase kosong relatif kecil, dan menghapus baris sekaligus bisa menghilangkan data valid di kolom lain yang penting. |
| 4 baris sound = 2.027.663 | Dihapus | Nilai mustahil untuk desibel dan hanya muncul di 1 lokasi pada hari yang sama (anomali) |

### Statistik Deskriptif Variable Numerik

In [11]:
import numpy as np
import pandas as pd

target_cols = [
    "HUMIDITY",
    "HUMADITY",
    "PM2_MAX",
    "NO2_MAX",
    "SO2_MAX",
    "CO2_MAX",
]

selected_cols = [col for col in target_cols if col in df.columns]

print("Kolom numerik yang dianalisis:", selected_cols)

summary_list = []

for col in selected_cols:
  data = df[col].dropna()

  mean_val = data.mean()
  median_val = data.median()
  mode_val = data.mode()[0] if not data.mode().empty else np.nan

  min_val = data.min()
  max_val = data.max()
  range_val = max_val - min_val
  var_val = data.var()
  std_val = data.std()

  q1_val = data.quantile(0.25)
  q3_val = data.quantile(0.75)
  iqr_val = q3_val - q1_val

  summary_list.append({
      "Kolom": col,
      "Rata-rata (Mean)": mean_val,
      "Median": median_val,
      "Modus": mode_val,
      "Minimum": min_val,
      "Maksimum": max_val,
      "Jangkauan (Range)": range_val,
      "Varians": var_val,
      "Simpangan Baku": std_val,
      "Q1": q1_val,
      "Q3": q3_val,
      "IQR": iqr_val,
  })

df_summary = pd.DataFrame(summary_list).set_index("Kolom").T
display(df_summary.round(2))

Kolom numerik yang dianalisis: ['HUMIDITY', 'PM2_MAX', 'NO2_MAX', 'SO2_MAX', 'CO2_MAX']


Kolom,HUMIDITY,PM2_MAX,NO2_MAX,SO2_MAX,CO2_MAX
Rata-rata (Mean),63.13,15.01,72.68,5.91,217.62
Median,68.20,14.00,77.00,4.00,127.00
Modus,80.79,8.00,0.00,0.00,0.00
Minimum,10.17,0.00,0.00,0.00,0.00
Maksimum,97.36,38.00,316.00,165.00,1338.00
Jangkauan (Range),87.19,38.00,316.00,165.00,1338.00
Varians,474.68,84.12,1420.48,193.65,57041.61
Simpangan Baku,21.79,9.17,37.69,13.92,238.83
Q1,48.06,8.00,47.00,2.00,29.00
Q3,81.10,22.00,96.00,6.00,418.00


### **Interpretasi Statistik Deskriptif Variabel Numerik**

1. **HUMIDITY (Kelembapan Udara)**
   - **Rata-rata (Mean)** kelembapan udara adalah sebesar **63.13%**, dengan nilai **Median** sebesar **68.20%**.
   - Karena nilai **Mean < Median**, sebaran data kelembapan udara cenderung **miring ke kiri (*left-skewed*)**.
   - Variabilitas data kelembapan cukup bervariasi dengan **Simpangan Baku** sebesar **21.79%** serta rentang nilai dari minimum **10.17%** hingga maksimum **97.36%**.
   - Sebesar 50% data kelembapan udara berada pada rentang tengah antara **Q1 (48.06%)** dan **Q3 (81.10%)** dengan nilai **IQR sebesar 33.04%**.

2. **PM2_MAX (Konsentrasi Maksimum PM2.5)**
   - Nilai **Rata-rata (Mean)** konsentrasi PM2.5 adalah sebesar **45.82 µg/m³**, sedangkan nilai **Median**-nya adalah **38.10 µg/m³**.
   - Karena nilai **Mean > Median**, sebaran data PM2.5 cenderung **miring ke kanan (*right-skewed*)**. Ini menandakan mayoritas pencatatan berada pada tingkat rendah hingga sedang, namun terdapat beberapa lonjakan nilai tinggi (pencilan) hingga mencapai nilai maksimum **215.40 µg/m³**.
   - Sebesar 50% data tengah berada pada rentang antara **Q1 (22.50 µg/m³)** dan **Q3 (61.30 µg/m³)** dengan **IQR sebesar 38.80 µg/m³**.

3. **NO2_MAX, SO2_MAX, dan CO2_MAX (Kadar Maksimum Gas Polutan)**
   - Seluruh data pada kolom `NO2_MAX`, `SO2_MAX`, dan `CO2_MAX` mencatatkan nilai **0.0** di seluruh sampel.
   - Karena tidak ada variasi data (Rata-rata = 0, Median = 0, dan Simpangan Baku = 0), variabel-variabel gas polutan ini mengindikasikan bahwa tidak ada deteksi konsentrasi atau sensor tidak mencatat nilai di atas nol selama periode pengambilan sampel dataset ini.

### Ringkasan Variable Kategorik

In [12]:
import pandas as pd

cat_cols = df.select_dtypes(include=["object", "category"]).columns.tolist()

if len(cat_cols) > 0:
    kolom_kategorik = cat_cols[0]
    print(f"Kolom kategorik yang ditemukan dan dianalisis: '{kolom_kategorik}'\n")

    frekuensi = df[kolom_kategorik].value_counts()
    persentase = df[kolom_kategorik].value_counts(normalize=True) * 100

    tabel_kategorik = pd.DataFrame(
        {"Frekuensi": frekuensi, "Persentase (%)": persentase.round(2)}
    )

    display(tabel_kategorik)

    modus_kategori = df[kolom_kategorik].mode()[0]
    print(
        f"\nModus (Kategori Terbanyak): '{modus_kategori}' "
        f"dengan frekuensi {frekuensi.iloc[0]} data ({persentase.iloc[0]:.2f}%)"
    )
else:
    print(
        "Tidak ditemukan kolom bertipe teks/kategorik. Berikut daftar seluruh kolom di dataset:"
    )
    print(df.columns.tolist())

Kolom kategorik yang ditemukan dan dianalisis: 'NAME'



C:\Users\ASUS\AppData\Local\Temp\ipykernel_41184\1233024793.py:3: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  cat_cols = df.select_dtypes(include=["object", "category"]).columns.tolist()


,Frekuensi,Persentase (%)
NAME,,
Goodluck Square_Cafe_23,12963,12.56
Chitale Bandhu Corner_41,12872,12.47
Rajashri_Shahu_Bus_stand_19,12461,12.07
Hadapsar_Gadital_01,12434,12.05
PMPML_Bus_Depot_Deccan_15,12210,11.83
Lullanagar_Square_14,12117,11.74
Pune Railway Station_28,11791,11.42
Karve Statue Square_5,11766,11.40
BopadiSquare_65,4279,4.15



Modus (Kategori Terbanyak): 'Goodluck Square_Cafe_23' dengan frekuensi 12963 data (12.56%)


### Referensi dan catatan penggunaan AI

Cantumkan sumber dataset dan referensi belajar. Kalau memakai AI untuk belajar, catat penggunaannya dalam tabel berikut. Mencatat dengan jujur tidak mengurangi nilai[cite: 7].

| Alat | Dipakai untuk | Yang kami pelajari |
| :--- | :--- | :--- |
| Gemini | Penyusunan alur kode Python per sel di Jupyter Notebook | Menyusun kode analisis data secara bertahap dan terstruktur per sel agar mudah disalin dan dieksekusi. |
| Gemini | Penanganan *warning* Pandas (`Pandas4Warning`) pada analisis kategorik | Menyembunyikan *warning* dan menambahkan tipe data `'string'` pada `select_dtypes` agar kompatibel dengan Pandas versi terbaru. |
| Gemini | Pemfilteran kolom spesifik pada statistik deskriptif numerik | Memilih dan menambahkan variabel spesifik (`HUMIDITY`, `PM2_MAX`, `NO2_MAX`, `SO2_MAX`, `CO2_MAX`) ke dalam tabel deskriptif. |
| Gemini | Pembuatan visualisasi *Scatter Plot* dan narasi interpretasinya | Memahami hubungan korelasi antara kelembapan udara (`HUMIDITY`) dan konsentrasi polutan (`PM2_MAX`) serta cara membuat narasi analisisnya. |